x-pack/plugins/transform/public/app/components/toast_notification_text.tsx

```diff
@@ -19,41 +19,43 @@ import {
 
 import { i18n } from '@kbn/i18n';
 
-import { CoreStart } from '@kbn/core/public';
+import { toMountPoint } from '@kbn/react-kibana-mount';
 
-import { toMountPoint } from '@kbn/kibana-react-plugin/public';
+import { useAppDependencies } from '../app_dependencies';
 
 const MAX_SIMPLE_MESSAGE_LENGTH = 140;
 
-// Because of the use of `toMountPoint`, `useKibanaContext` doesn't work via `useAppDependencies`.
-// That's why we need to pass in `overlays` as a prop cannot get it via context.
 interface ToastNotificationTextProps {
-  overlays: CoreStart['overlays'];
-  theme: CoreStart['theme'];
   text: any;
   previewTextLength?: number;
+  inline?: boolean;
+  forceModal?: boolean;
 }
 
 export const ToastNotificationText: FC<ToastNotificationTextProps> = ({
-  overlays,
   text,
-  theme,
   previewTextLength,
+  inline = false,
+  forceModal = false,
 }) => {
-  if (typeof text === 'string' && text.length <= MAX_SIMPLE_MESSAGE_LENGTH) {
+  const { overlays, theme, i18n: i18nStart } = useAppDependencies();
+
+  if (!forceModal && typeof text === 'string' && text.length <= MAX_SIMPLE_MESSAGE_LENGTH) {
     return text;
   }
 
   if (
+    !forceModal &&
     typeof text === 'object' &&
     typeof text.message === 'string' &&
     text.message.length <= MAX_SIMPLE_MESSAGE_LENGTH
   ) {
     return text.message;
   }
 
-  const unformattedText = text.message ? text.message : text;
-  const formattedText = typeof unformattedText === 'object' ? JSON.stringify(text, null, 2) : text;
+  const unformattedText = typeof text === 'object' && text.message ? text.message : text;
```

Comment:
Could `text` be null here? typeof null is 'object'.

Reply:
Fixed in f25ea75b32e.